safe_transaction_service/utils/admin.py

```diff
@@ -0,0 +1,35 @@
+from typing import Tuple
+
+from django.contrib import admin
+from django.core.exceptions import ValidationError
+from django.db.models import QuerySet
+from django.http import HttpRequest
+
+
+class BinarySearchAdmin(admin.ModelAdmin):
+    def get_search_results(
+        self, request: HttpRequest, queryset: QuerySet, search_term: str
+    ) -> Tuple[QuerySet, bool]:
+        queryset, may_have_duplicates = super().get_search_results(
+            request, queryset, search_term
+        )
+        if search_term:
+            for search_field in self.get_search_fields(request):
+                try:
+                    if search_field.startswith("="):
+                        may_have_duplicates = True
```

Comment:
Why do we change the `may_have_duplicates` here and not take the one from `super().get_search_results()`?